go/vt/sqlparser/token.go

```diff
@@ -661,7 +661,14 @@ func (tkn *Tokenizer) Scan() (int, []byte) {
 		}
 		return tkn.scanIdentifier(byte(ch), isDbSystemVariable)
 	case isDigit(ch):
-		return tkn.scanNumber(false)
+		typ, res := tkn.scanNumber(false)
+		if typ != LEX_ERROR {
```

Comment:
Need tests that invalid numbers still generate a parse error in places they can't be used as an identifier, and that the error makes sense

Things like `select 123a from dual`